example/kotlinlib/basic/7-dependency-injection/build.mill

```diff
@@ -0,0 +1,89 @@
+//// SNIPPET:BUILD
+
+package build
+import mill._, kotlinlib._
+import kotlinlib.ksp.KspModule
+
+object dagger extends KspModule { outer =>
+
+  def kotlinVersion = "2.1.0"
+
+  def symbolProcessingVersion = "1.0.29"
+
+  def kotlinLanguageVersion = "1.9"
+
+  def mainClass = Some("com.example.dagger.MainKt")
+
+  def kotlincOptions = super.kotlincOptions() ++ Seq("-no-reflect", "-verbose")
+
+  def kotlinSymbolProcessing = true
+
+  def ivyDeps = Agg(
+    ivy"com.google.dagger:dagger-compiler:2.55"
+  )
+
+  def kotlinSymbolProcessors = Agg(
+    ivy"com.google.dagger:dagger-compiler:2.55"
+  )
+
+  object test extends KotlinTests with TestModule.Junit5 with KspModule {
```

Comment:
If using KSP in test modules is the default or needed often, we should think about providing a pre-configured `KspTests` trait, so we don't need to refer to `outer` to make consistent settings.